src/engine/framework/VirtualMachine.cpp

```diff
@@ -246,54 +275,111 @@ static std::pair<Sys::OSHandle, IPC::Socket> CreateNaClVM(std::pair<IPC::Socket,
 		Log::Warn("NaCl loader not found: %s", nacl_loader);
 	if (!FS::RawPath::FileExists(irt))
 		Log::Warn("NaCl integrated runtime not found: %s", irt);
-#ifdef __linux__
-	#if defined(DAEMON_ARCH_arm64)
+
+#if defined(__linux__) || defined(__FreeBSD__)
+	bool enableBootstrap = vm_nacl_bootstrap.Get();
+
+	if (enableBootstrap)
+	{
+#if defined(DAEMON_ARCH_arm64)
 		bootstrap = FS::Path::Build(naclPath, "nacl_helper_bootstrap-armhf");
-	#else
+#else
 		bootstrap = FS::Path::Build(naclPath, "nacl_helper_bootstrap");
-	#endif
-	if (!FS::RawPath::FileExists(bootstrap))
-		Log::Warn("NaCl bootstrap helper not found: %s", bootstrap);
-	args.push_back(bootstrap.c_str());
+#endif
+
+		if (!FS::RawPath::FileExists(bootstrap))
```

Comment:
I see that this code was already there, but it ought to be a Sys::Error if part of the loader is missing.